canvas_site_creator/views.py

```diff
@@ -53,3 +269,116 @@ def create_new_course(request):
     return render(request, 'canvas_site_creator/create_new_course.html',
                   {'school_id': school_id, 'school_name': school.title_short,
                    'canvas_site_templates': canvas_site_templates})
+
+@login_required
+@lti_permission_required(canvas_api_accounts.ACCOUNT_PERMISSION_MANAGE_COURSES)
+@require_http_methods(['GET'])
+def partials(request, path):
+    return render(request, 'canvas_site_creator/partials/' + path, {})
+
+
+@login_required
+@lti_permission_required(canvas_api_accounts.ACCOUNT_PERMISSION_MANAGE_COURSES)
+@require_http_methods(['POST'])
+def create_job(request):
+    canvas_user_id = request.LTI['custom_canvas_user_id']
+    logged_in_user_id = request.LTI['lis_person_sourcedid']
+    data = json.loads(request.POST['data'])
+
+    template_canvas_course_id = data.get('template')
+    filters = data['filters']
+    term = filters.get('term')
+
+    school_account_id = filters['school']
+    (account_type, school_id) = canvas_api_accounts.parse_canvas_account_id(school_account_id)
+
+    department = None
+    department_account_id = filters.get('department')
+    if department_account_id:
+        (account_type, department) = department_account_id.split(':')
+
+    course_group = None
+    course_group_account_id = filters.get('course_group')
+    if course_group_account_id:
+        (account_type, course_group) = course_group_account_id.split(':')
+
+    created_by_user_id = logged_in_user_id
+    if not created_by_user_id:
+        created_by_user_id = "canvas_user_id:%s" % canvas_user_id
+
+    # If the create_all flag has been set and passed with the form,
+    # then create a query to get the all course instances with the applied filters that are to be created.
+    course_instance_ids = []
+    if data.get('create_all', False):
+        # Get the account data to be used in the course instance query.
+        if filters['course_group']:
+            account = filters['course_group']
+        elif filters['department']:
+            account = filters['department']
+        else:
+            account = filters['school']
+
+        # Retrieve all course instances for the given term and account that do not have Canvas course sites
+        # nor are set to be fed into Canvas via the automated feed
+        ci_query_set_without_canvas = get_course_instance_query_set(term, account).filter(
+            canvas_course_id__isnull=True, sync_to_canvas=0)
+
+        # Iterate through the query set to build a list of all the course instance id's
+        # for a school/course_group/department, which course sites will be created for.
+        for ci in ci_query_set_without_canvas:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To create a `list`, try to use `list` comprehension instead of a loop. List comprehension is the preferred way to make a list using Python, and it&#39;s simpler and easier to understand than using a loop.

[Learn more](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions)